### ISSUE TITLE
How does SeiT/SeiT++ perform semantic segmentation tasks?

### ISSUE THREAD

**Issue by acrlife** [naver-ai/seit#3]
Hi @SanghyukChun @8uos @gaviotas 
I am planning to use SeiT for semantic segmentation tasks as described in your paper, but I have some uncertainties that I hope you can clarify:

‌Tokenizer and Codebook Configuration‌:
Does the ADE20K dataset use ‌exactly the same tokenizer and codebook‌ as ImageNet-1k for token extraction? 
Specifically, do the tokenizer and codebook need to be ‌retrained‌ for ADE20K?
Does the codebook size remain ‌391 codewords‌ (as in ImageNet-1k)?
After resizing ADE20K images to ‌512×512‌, are they partitioned into ‌non-overlapping 8×8 patches‌?
‌
Ground Truth Processing‌:
Does the ADE20K ground truth (semantic labels) need to be converted into ‌token-level representations‌ (e.g., mapped to codebook indices)?

‌Data Loading and Augmentation in MMSegmentation‌:
What are the ‌default data loading and augmentation strategies‌ implemented in MMSegmentation for SeiT?
If possible, could you provide details about the ‌data pipeline configuration‌ (e.g., preprocessing steps, batching, augmentation policies)?
I would greatly appreciate your guidance on these questions, especially regarding the ‌MMSegmentation data loading pipeline‌.

++++++++++++++++++++++++++++++++++++++++++++++++++

Here is the code I wrote to extract token data and its index files from ADE20K. In order to be able to use your tokenizer.ckpt and codebook.ckpt, I set the patch size to 16.
May I ask if my method is correct?

`
import os
import gzip
import numpy as np
from PIL import Image
import torch
from tqdm import tqdm
from torchvision import transforms
import sys
from types import SimpleNamespace
sys.path.append("D:/GithubProject/enhancing-transformers(ViT-VQGAN)")

from token_transform import build_codebook

def generate_ade20k_tokens(
    ade20k_root: str,
    encoder_ckpt: str,
    codebook_path: str,
    output_dir: str,
    img_size: int = 512,
    patch_size: int = 16,
    device='cuda' if torch.cuda.is_available() else 'cpu'
):
    
    os.makedirs(os.path.join(output_dir, 'tokens'), exist_ok=True)
    os.makedirs(os.path.join(output_dir, 'index'), exist_ok=True)

    
    encoder = torch.load(encoder_ckpt)
    original_conv = encoder[0].to_patch_embedding[0]
    new_conv = torch.nn.Conv2d(
        in_channels=3,
        out_channels=original_conv.out_channels,
        kernel_size=patch_size,  
        stride=patch_size,       
        padding=0
    )
    encoder[0].to_patch_embedding[0] = new_conv
    encoder = encoder.to(device)  
    
    encoder.eval()

    args = SimpleNamespace(codebook_path=codebook_path)  # Create proper args object
    codebook, num_tokens = build_codebook(args)
    codebook = codebook.weight.data.to(device) 
    num_codes, code_dim = codebook.shape

    transform = transforms.Compose([
        transforms.Resize((img_size, img_size)),
        transforms.ToTensor(),
        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
    ])

    img_dir = os.path.join(ade20k_root, 'images', 'training')
    img_paths = [os.path.join(img_dir, f) for f in os.listdir(img_dir) if f.endswith('.jpg')]

    token_data = bytearray()
    index_list = [0]  

    for img_path in tqdm(img_paths, desc="Generating ADE-20k Tokens"):
        img = Image.open(img_path).convert('RGB')
        img_tensor = transform(img).unsqueeze(0).to(device)  # [1, 3, 512, 512]

        with torch.no_grad():
            encoded_vectors = encoder(img_tensor) 
        encoded_vectors = encoded_vectors.reshape(-1, code_dim)  # [1024, 32]

        distances = torch.cdist(encoded_vectors, codebook)  # [1024, num_codes]
        code_indices = distances.argmin(dim=1).cpu()  # [1024]
        tokens = code_indices.reshape(32,32)  # [32, 32]

        tokens_np = tokens.numpy().astype(np.uint16)
        token_flat = tokens_np.flatten().tobytes()  
        token_flat = tokens_np.flatten().tobytes() 

     
        token_data.extend(token_flat)
        index_list.append(index_list[-1] + len(token_flat))  

    
    token_save_path = os.path.join(output_dir, 'tokens', 'ade20k_512-val-token.data.bin.gz')
    with gzip.open(token_save_path, 'wb') as f:
        f.write(token_data)

    index_save_path = os.path.join(output_dir, 'index', 'ade20k_512-val-token.index.npy.gz')
    with gzip.open(index_save_path, 'wb') as f:
        np.save(f, np.array(index_list, dtype=np.int64))

    print(f"Generate ADE-20k tokens：\n- Data files: {token_save_path}\n- Index files: {index_save_path}")
`
